codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java

```diff
@@ -0,0 +1,237 @@
+    private final Map<AttributeKey<?>, Object> attributes;
+
+    // Lock for parentChannel and multiplexCodec
+    private final Object lock = new Object();
+
+    public Http2StreamChannelBootstrap() {
+        options = new LinkedHashMap<ChannelOption<?>, Object>();
+        attributes = new LinkedHashMap<AttributeKey<?>, Object>();
+    }
+
+    // Copy constructor
+    Http2StreamChannelBootstrap(Http2StreamChannelBootstrap bootstrap0) {
+        synchronized (bootstrap0.lock) {
+            parentChannel = bootstrap0.parentChannel;
+            multiplexCodec = bootstrap0.multiplexCodec;
+        }
+        handler = bootstrap0.handler;
+        group = bootstrap0.group;
+        synchronized (bootstrap0.options) {
+            options = new LinkedHashMap<ChannelOption<?>, Object>(bootstrap0.options);
+        }
+        synchronized (bootstrap0.attributes) {
+            attributes = new LinkedHashMap<AttributeKey<?>, Object>(bootstrap0.attributes);
+        }
+    }
+
+    /**
+     * Creates a new channel that will eventually map to a local/outbound HTTP/2 stream.
+     */
+    public ChannelFuture connect() {
+        return connect(-1);
+    }
+
+    /**
+     * Used by the {@link Http2MultiplexCodec} to instantiate incoming/remotely-created streams.
+     */
+    ChannelFuture connect(int streamId) {
+        validateState();
+
+        Channel parentChannel0;
+        Http2MultiplexCodec multiplexCodec0;
+        synchronized (lock) {
+            parentChannel0 = parentChannel;
+            multiplexCodec0 = multiplexCodec;
+        }
+        EventLoopGroup group0 = group;
+        group0 = group0 == null ? parentChannel.eventLoop() : group0;
+        ChannelHandler handler0 = handler;
+
+        return multiplexCodec0.createStreamChannel(parentChannel0, group0, handler0, options, attributes, streamId);
+    }
+
+    /**
+     * Sets the parent channel that must have the {@link Http2MultiplexCodec} in its pipeline.
+     *
+     * @param parent a registered channel with the {@link Http2MultiplexCodec} in its pipeline. This channel will
+     *               be the {@link Channel#parent()} of all channels created via {@link #connect()}.
+     * @return {@code this}
+     */
+    public Http2StreamChannelBootstrap parentChannel(Channel parent) {
+        synchronized (lock) {
+            parentChannel = checkRegistered(checkNotNull(parent, "parent"));
+            multiplexCodec = requireMultiplexCodec(parentChannel.pipeline());
+        }
+        return this;
+    }
+
+    /**
+     * Sets the channel handler that should be added to the channels's pipeline.
+     *
+     * @param handler   the channel handler to add to the channel's pipeline. The handler must be
+     *                  {@link Sharable}.
+     * @return {@code this}
+     */
+    public Http2StreamChannelBootstrap handler(ChannelHandler handler) {
+        this.handler = checkSharable(checkNotNull(handler, "handler"));
```

Comment:
do you see practical use cases where this may need to change dynamically? is it sufficient just to make this `final` for now and set it in the constructor?